rmw_cyclonedds_cpp/src/rmw_node.cpp

```diff
@@ -2492,6 +2759,11 @@ static rmw_ret_t destroy_subscription(rmw_subscription_t * subscription)
   rmw_ret_t ret = RMW_RET_OK;
   auto sub = static_cast<CddsSubscription *>(subscription->data);
   clean_waitset_caches();
+
+  dds_listener_t * listener = dds_create_listener(nullptr);
```

Comment:
These 3 lines have no functional effect, I'd delete them. The same is true in `destroy_client` and `destroy_service`.